### ISSUE TITLE
Post a Problem without Disqualifying the beatmap

### ISSUE THREAD

**Issue by Loctav** [ppy/osu-web#4405]
Refer to https://github.com/ppy/osu-web/issues/4404

Current behavior causes users with the permissions to disqualify a map to be unable to post a Problem on a beatmap discussion without disqualifying it, as the button for "Problem" gets replaced with "Disqualify". (https://puu.sh/Do7wY/3bae53a0bb.png)

In order to have at least a second Beatmap Nominator be able to read about and agree to a potential problem that leads to a Disqualification of a beatmap, the following change is requested:

* Make it possible for users with the permission to Disqualify a map to post an issue with the "Problem" flag without actually disqualifying it.

**Comment by Mordredosu** [comment]
completely agree with this, problems should not insta dq for reasons loctav already stated

apparently bns have 2 dq buttons as you can see here https://i.imgur.com/A3FRMun.png so imo the easiest solution would just be to keep the dq button below hype, which would just dq the map, and replace the disqualify button at the bottom with problem again

that way bns / nats / qats would be able to keep posting problems on qualified maps without disqualifying the map; 

just like how nominating a map doesn't post a message in the map thread, disqualifying wouldn't do that either, instead bns would be required to post a problem stamp after disqualifying a map, stating the reasoning

**Comment by Mordredosu** [comment]
only issue is that you don't have that button on your own maps, see https://hypercyte.s-ul.eu/kP90WMb1 this for example, so you might have to do something about that idk

**Comment by cl8n** [comment]
the problem posting is used so that every disqualification is tied to a reason

**Comment by nanaya** [comment]
If it's not quite a problem then maybe post as suggestion? Then if it turns out to be a problem, someone else can then refer to that in disqualification post?

Also the disqualify button at top just sends you to the new discussion box. There's logic mismatch between the new post and nomination box which needs fixing though (you can't disqualify your own map at the moment).

**Comment by cl8n** [comment]
yeah I think this is best practice to use. if it's an obvious issue (e.g. unsnapped notes, ...) they can post as problem to DQ, and if the problem needs some other discussion it can be a suggestion at first

**Comment by Mordredosu** [comment]
bn rules don't allow dq'ing without reporting the map first, meaning you can't post a problem for that

problem vs suggestions also have a completely different impression on the mapper, currently you're forced to post something as a suggestion no matter how serious the issue might be

I still think either adding a new button or just creating a prompt asking "do you want to dq the map with this" would be the best solution, though I have absolutely no experience with coding so I don't know how hard that is to implement

**Comment by cl8n** [comment]
if it's a serious issue that a BN is sure of why not DQ immediately? I guess I am suggesting a change to BN rules if something contradictory is written there

**Comment by Loctav** [comment]
I disagree as a suggestion is more like a request to have something changed to your personal preferences, whereas problems are actual problems deattached from your personal preferences. Something can be a potential problem (and should be posted as problem) while it is not a suggestion at all. Right now, you can not post a problem (aka opening the can of worms to discuss the severety of a problem and whether or not you are wrong with it) without DQing it and posting a suggestion (you are not suggesting things, you want to point out a real problem here and you want to talk it out first) is not really doing the trick here.

**Comment by cl8n** [comment]
so I guess where I'm confused is why you would not want to disqualify if you found an "actual problem"? it kinda makes sense to me that a map with a problem would be disqualified

**Comment by VenixOSU** [comment]
Suggestion is more like personal or minor thing which is acceptable in its current state, while there are things like timing, metadata, spread and so on which might need more discussion/opinions or mapper-side clarification (e.g. metadata might be taken from other references, timing might be generalizing something for some reason, etc.), those things aren't just suggestions but potential issues which shouldn't require direct disqualification. Not every issue is as objective as unsnapped objects or unused files (in such cases direct DQ would be obvious).

**Comment by Naxesss** [comment]
In addition to what Venix said, getting an extra pair of eyes to confirm that the disqualification is not a mistake is a pretty quick thing to do and mitigates the risk, and so it is something that we have as part of the process. If the issue really is obvious, confirming it will be equally obvious. It's easy to say that if a nominator disqualifies by mistake it can be fixed, but the damage to the trust mappers have in that disqualifications are valid can't be undone, and it also causes extra work and stress for everyone else (nominators who nominated, members of the nomination assessment team who have to clear up the situation, the nominator who disqualified, and the mappers who may have forced themselves to make detrimental changes) which could have easily been avoided.

If the map is close to ranked, we loosen the process requirements to ensure that it never gets in the way of actually stopping maps before it's too late, but while there's time, double checking is more beneficial than it is detrimental.

At the moment, nominators need to resort to explicitly saying that something is a problem when they set the suggestion label, in order to express their intention with the post, which contradicts the purpose of labels.

![image](https://user-images.githubusercontent.com/30292137/57442890-35130d00-724d-11e9-8c40-0ca9457a1898.png)

**Comment by nanaya** [comment]
So it's _possibly_ a problem and someone needs to confirm it by posting a disqualifying post pointing to that...?

**Comment by Naxesss** [comment]
That's how it works currently yeah, but ideally you'd be able to just disqualify it without posting anything and then linking to the post with the problem stamp in the disqualify message.

**Comment by cl8n** [comment]
> the damage to the trust mappers have in that disqualifications are valid can't be undone

where does this interpretation of DQs come from? I don't think they're supposed to be *proof* that a map has problems. they give time for more people, including the mapper, to look into likely problems brought up or supported by BNs/NAT

> and it also causes extra work and stress for everyone else

again I feel like this is just applying an old stigma about DQs to the modern system where it really doesn't fit. what is stressful about a DQ? why would you want a map where a BN thinks there is a problem to be sitting in Qualified and on its way toward Ranked?

**Comment by Naxesss** [comment]
> where does this interpretation of DQs come from?

the Beatmap Ranking Procedure and Beatmap Nominator Rules, which both state that DQs happen due to rule violations. they are not proof, but newer mappers and modders rely on them to make decisions (certain parts of the rc get pretty extensive and requires experience, like timing/metadata)

> what is stressful about a DQ?

mappers need to make potentially unwanted changes to the map, for example changing 1/4 sections to 1/3 in lower difficulties, even though experienced bns would say it works as simplification. nominating BNs may lose their position depending on severity. this becomes especially stressful for anyone on probation. questionable DQs are also generally the only ones that attract drama where gmt need to get involved

spending the time to just get another person to double check within minutes usually helps way more than it is detrimental, hence why it's part of the procedure. if a map is close to rank, which is really the only situation this part may be bad, then it becomes optional

I wouldn't be against removing it, but I don't think it's a good idea to do so. you save maybe a few minutes, at best hours, DQing a map that wouldn't get ranked in multiple days for something that at worst shouldn't even be DQed

**Comment by cl8n** [comment]
this seems silly to me that DQs are treated like their own type of post; they're problem posts, nothing more. a qualified map shouldn't have unresolved problems (problems verified as such by a BN, fwiw). I'm curious what the modding community's feelings towards DQs would look like if DQs weren't given so much fake importance via certain BN rules and BN evals and a lingering association to forum-modding DQs etc

**Comment by Okorin** [comment]
a map getting disqualified is a major pain in the butt, you'll need to get the people involved in it previously to verify whatever was changed and nominate again, obviously you'd only want to go through the hassle if the problem that someone posted wasn't an error on their part / actually a problem that needs fixing

dqs don't waste as much time anymore, but coordination with people does waste a bunch of time inbetween so a set of behaviours was put in place to minimize time wasted with stuff not worth discussing

a problem that carries weight to disqualify kind of is different than _just_ the problem posted by someone by resulting in different outcomes based on your usergroup they're treated like their own thing because they result in their own thing

**Comment by celerih** [comment]
Would be nice to not be so quickly dismissive of problems brought up by the modding community - _______ - @clayton

DQs are their own kind of post, they affect the map's status and for the longest time the system has worked on a report -> DQ system to avoid errors where possible or give more time for people to discuss issues before the map's status is changed. This is something that is very clearly needed, since often issues brought up are later discussed to be fine, or better solutions are found, which is two things we wouldn't get by just going right for the DQ and rushing to requal. DQs have a certain importance to them because they affect not just it's status in qualified but also in the map dqs bring multiple changes. Taking it more slowly with them is the better option.

So please just give this option to BNs already

**Comment by peppy** [comment]
If disqualification is not the result of reporting a problem, there is potential for it to *continue to a ranked state before it's even addressed*.

A map getting disqualified SHOULD BE A SMALL PAIN IN THE BUTT. The idea is to ensure quality, not ensure everyone gets a smooth ride to ranking. We've already been through this. We've already eased the pains surrounding disqualification.

All problems deserve disqualification. If not, it's not a problem. Keep things simple.
